backend/tracim_backend/lib/translate/providers.py

```diff
@@ -0,0 +1,41 @@
+from enum import Enum
+import typing
+
+from sqlalchemy.orm import Session
+
+from tracim_backend.lib.translate.services.systran import SystranTranslationService
+from tracim_backend.lib.translate.translator import TranslationService
+from tracim_backend.models.auth import User
+
+if typing.TYPE_CHECKING:
+    from tracim_backend.config import CFG
+
+
+class TranslationProvider(str, Enum):
+    SYSTRAN = "systran"
+
+
+TRANSLATION_SERVICE_CLASSES = {TranslationProvider.SYSTRAN: SystranTranslationService}
+
+
+class TranslationLib:
+    """
+    Tracim Lib to use TranslationServices
+    """
+
+    def __init__(
+        self, session: Session, current_user: typing.Optional[User], config: "CFG"
+    ) -> None:
+        self._user = current_user
+        self._session = session
+        self._config = config
+
+    def get_translation_service(self) -> typing.Optional[TranslationService]:
+
+        if self._config.TRANSLATION_SERVICE__PROVIDER == TranslationProvider.SYSTRAN:
```

Comment:
Wouldn't it be possible to store the provider's specific settings in a dict (`self._config.TRANSLATION_SERVICE__SETTINGS`)?
Then a code like that would be generic:
```python
try:
    provider_class = TRANSLATION_SERVICE_CLASSES[self._config.TRANSLATION_SERVICE__PROVIDER]
    provider_settings = self._config.TRANSLATION_SERVICE__PROVIDER_SETTINGS
    return provider_class(**provider_settings)
except KeyError:
    # I think we should raise an error, or do nothing to catch the exception
    raise
```

Reply:
seems complicated.